### ISSUE TITLE
How do i make it transfer certain pokemon like in the old version

### ISSUE THREAD

**Issue by yehan77** [PokemonGoF/PokemonGo-Bot#3206]
Please check configuration at http://jsonlint.com/ before posting an issue.

### Expected Behavior
I want to have the bot transfer certain pokemon like in the old version where i could type the lowest cp and anything under would be transfered

### Actual Behavior
it doesn't transfer anything

### Steps to Reproduce

2016-08-09 14:02:02,592 [PokemonCatchWorker] [INFO] [pokemon_appeared] A wild Poliwag appeared! [CP 438] [Potential 0.62] [S/A/D 2/14/12]
2016-08-09 14:02:05,929 [PokemonCatchWorker] [INFO] [threw_berry] Threw a Razz Berry! Catch rate now: 47.88
2016-08-09 14:02:05,929 [PokemonCatchWorker] [INFO] [threw_pokeball] Used Pokeball, with chance 47.88 (80 left)
2016-08-09 14:02:07,709 [PokemonCatchWorker] [INFO] [pokemon_caught] Captured Poliwag! [CP 438] [Potential 0.62] [2/14/12] [+210 exp]
### Other Information
OS: OSX
Git Commit: (run 'git log -n 1 --pretty=format:"%H"' and paste it here)
Python Version: (run 'python -V' and paste it here)

**Comment by shivamkalra** [comment]
Did you check the release check in config example? It cannot be clear than this:

https://github.com/PokemonGoF/PokemonGo-Bot/blob/master/configs/config.json.example#L96

For your case:
```
"release": {
      "any": {"release_below_cp": 200},
      ....
    },```

**Comment by yehan77** [comment]
With the old one it had a whole config with all the pokemon names that i could type the cp to release, is it possible to be the same with this one? there is only a hand full of names here not like all of them.

**Comment by Maxigon** [comment]
look for the config.json.pokemon.example in /configs

**Comment by raxxon** [comment]
It seems that it won't go back through captured pokemon and release if I stop the bot and update the release configuration. I can see it releasing newly captured/evolved pokemon, but not actually pruning the existing pokemon pool....

**Comment by Baconnaise** [comment]
You can look at the example config.json files in the config folder. I've attached an example of this as well. The old way had a separate configuration which was replaced. The bot now uses the release configuration which is more efficient and easier to fine tune.

Here is an example of the default release config. 
![example](https://cloud.githubusercontent.com/assets/20616256/17507676/f705c280-5dde-11e6-9188-003771fd3c0e.JPG)

I have a CP 500 set and a 0.8 IV here. Now on the right it says "or". I can use "and" or "or". Since it says "or" that means any pokemon I catch that are under 500 CP and under 0.8 IV will be released. If I put "and" instead of "or" that means it will keep any pokemon above 500 CP or under 0.8.

![example1](https://cloud.githubusercontent.com/assets/20616256/17507664/ed979926-5dde-11e6-8127-802b57c57cb5.JPG)

**Comment by raxxon** [comment]
Config excerpt:

``` 
  "release": {
      "any": {"release_below_cp": 100, "release_below_iv": 0.5, "logic": "or" },
```

``` 
      "Caterpie": { "release_below_cp": 220, "release_below_iv": 0.8, "logic": "or" },
      "Weedle": { "release_below_cp": 250, "release_below_iv": 0.8, "logic": "or" },
      "Diglett": { "release_below_cp": 158, "release_below_iv": 0.8, "logic": "and" },
      "Metapod": { "release_below_cp": 275, "release_below_iv": 0.8, "logic": "or" },
      "Kakuna": { "release_below_cp": 220, "release_below_iv": 0.8, "logic": "or" },
      "Rattata": { "release_below_cp": 375, "release_below_iv": 0.8, "logic": "or" },
      "Abra": { "release_below_cp": 208, "release_below_iv": 0.8, "logic": "and" },
      "Zubat": { "release_below_cp": 350, "release_below_iv": 0.8, "logic": "or" },
```

Any NEW pokemon caught/evolved are subjected to the rules. Older pokemon are ignored. This wasn't the way it was. As the bot has leveled the CP ratings have increased. So over time I've stopped the bot, tuned the rules to "thin the herd".... the bot is no longer thinning the herd, it's ignoring currently owned pokemon.

and it's also not honoring new settings in the RecycleItems filter either, but that's a different issue.....

**Comment by Baconnaise** [comment]
If that's the case then it sounds like you need to tune the rules again. What you posted above means you're keeping everything that is 100CP or higher with an IV 0.5 or higher. It will prune anything above 100CP that doesn't meet the IV of 0.5 as well. This is besides what you have set below which will trump the above. Quite a wide net.

Are you using the latest dev or master?

**Comment by raxxon** [comment]
Latest Dev. I pulled before I went to work earlier today, updated the rules from remote and waited for the bot to crash and restart to load them. Noticed no change. Came home, shut down bot, did another pull (including requirement check/pgoapi update), updated the rules again and still no purging of old pokemon.

Until I did the recent pull I left the "any" at the defaults of 0 and 0. I've been updating individual pokemon as they're being collected as at times I will have 20 of one kind or another and really don't need that many "high level" Pidgeys. Some I've changed from AND to OR to purge them even harder and only keep the Best of the Best when they're found.

The key problem I'm seeing is that the release configurations are working, but for some reason are not being applied to pokemon that have already passed an old filter and were kept. This is not how it was working before the API change happened and shut things down for a bit. I know because the bot ran for several hours while I was asleep and wound up maxing out at one point, I updated configurations and restarted and before it started walking it was shipping off tons of Rats and Birds to the Great Woodchipper to be turned into candy...

**Comment by raxxon** [comment]
The problem I'm seeing is most likely not code-generated.

Using the new Linux on WIndows bollocks that came with Windows 10.1 there is a SEVERE limitation. Any file that is in the "Linux" environment can **NOT** be edited/saved by a Windows application. The program will not save all the extended attributes and will cause the files to vanish from the Linux environment. While the Linux environment is active the system has a cached copy of the filesystem in RAM, however once you exit bash everything is flushed to disk. The files are present on the NTFS filesystem but without the extended attributes it will not be visible from the Linux side.

Long story short, when I edited the configs in PN2 it vanished from the linux side, but a cached copy was still visible (and un-edited) until I exited the bash shells at which time everything vanished. Since I've been working from a cached copy of the config none of my changes were actually being written out where the bot could use them and therefor I screwed myself.....

**Comment by raxxon** [comment]
Nuked from orbit. Reinstalled lxss. Reinstalled pre-reqs and compiled crypto lib again. Manually restored configurations that were backed up. Bot working as expected, released all excess pokemans in the inventory and cleaned up items as well.  so much hate...

**Comment by Baconnaise** [comment]
One more reason to hate the Win10 botnet. If it helps you reinforced my decision to stay away from it even with my dual boot or VM.
